<html>
  <head>
	  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    <title>i5ting_ztree_toc:README</title>
		<link href="toc/style/github-bf51422f4bb36427d391e4b75a1daa083c2d840e.css" media="all" rel="stylesheet" type="text/css"/>
		<link href="toc/style/github2-d731afd4f624c99a4b19ad69f3083cd6d02b81d5.css" media="all" rel="stylesheet" type="text/css"/>
		<link href="toc/css/zTreeStyle/zTreeStyle.css" media="all" rel="stylesheet" type="text/css"/>
	  <style>
		pre {
		    counter-reset: line-numbering;
		    border: solid 1px #d9d9d9;
		    border-radius: 0;
		    background: #fff;
		    padding: 0;
		    line-height: 23px;
		    margin-bottom: 30px;
		    white-space: pre;
		    overflow-x: auto;
		    word-break: inherit;
		    word-wrap: inherit;
		}

		pre a::before {
		  content: counter(line-numbering);
		  counter-increment: line-numbering;
		  padding-right: 1em; /* space after numbers */
		  width: 25px;
		  text-align: right;
		  opacity: 0.7;
		  display: inline-block;
		  color: #aaa;
		  background: #eee;
		  margin-right: 16px;
		  padding: 2px 10px;
		  font-size: 13px;
		  -webkit-touch-callout: none;
		  -webkit-user-select: none;
		  -khtml-user-select: none;
		  -moz-user-select: none;
		  -ms-user-select: none;
		  user-select: none;
		}

		pre a:first-of-type::before {
		  padding-top: 10px;
		}

		pre a:last-of-type::before {
		  padding-bottom: 10px;
		}

		pre a:only-of-type::before {
		  padding: 10px;
		}

		.highlight { background-color: #ffffcc } /* RIGHT */
		</style>
  </head>
  <body>
	  <div>
				<div style='width:25%;'>
						<ul id="tree" class="ztree" style='width:100%'>

						</ul>
				</div>
        <div id='readme' style='width:70%;margin-left:20%;'>
          	<article class='markdown-body'>
            	<h1 id="gdbigtools-a-command-line-tools-for-gdbig-varaints-browser">GDBIGtools: A command line tools for GDBIG varaints browser</h1>
<p><a href="https://pypi.org/project/GDBIGtools/"><img src="https://img.shields.io/pypi/v/GDBIGtools.svg" alt="PyPI Version"></a>
<a href="https://github.com/BIGCS-Lab/GDBIGtools/blob/master/LICENSE"><img src="https://img.shields.io/pypi/l/GDBIGtools.svg" alt="License"></a></p>
<h2 id="introduction">Introduction</h2>
<p>The <a href="http://gdbig.bigcs.com.cn">Genome Database of Born In Guangzhou Cohort study(GDBIG)</a> is based on thousands of trios/duos families recruited by the BIGCS Project to conduct whole-genome-sequencing, genome variation detection, annotation and analysis. Phase I including 332 trios (father-mother-offspring), 1,406 duos (14 father-offspring and 1,392 mother-offspring) and 245 single individuals (70 infants, 150 adult females and 25 adult males), for a total of 4,053 individual samples. The GDBIG delivers periodical and useful variation information and scientific insights derived from the analysis of thousands of born in Guangzhou China sequencing data. The results aim to promote genetic research and precision medicine actions in China. The delivering information includes any of detected variants and the corresponding allele frequency, annotation, frequency comparison to the global populations from existing databases, etc.</p>
<p>GDBIG variant browser allows authorized access its data through an Genomics API and <strong>GDBIGtools</strong> is a convenient command line tools for the <a href="http://gdbig.bigcs.com.cn">GDBIG</a> variants browser.</p>
<p>No only GDBIG database but also other external databases\&#39; Allele Frequency you could abtain.
Up to now, the following databases are integrated:</p>
<ul>
<li><a href="http://gdbig.bigcs.com.cn">GDBIG</a></li>
<li><a href="https://linkinghub.elsevier.com/retrieve/pii/S0092867418310328">CMDB</a></li>
<li><a href="http://www.mbiobank.com/">ChinaMAP</a></li>
<li><a href="http://bigdata.ibp.ac.cn/NyuWa_variants/">NyuWa (NCVD)</a></li>
<li><a href="https://wbbc.westlake.edu.cn/">WBBC</a></li>
<li><a href="https://gnomad.broadinstitute.org">gnomAD</a></li>
<li><a href="https://www.internationalgenome.org/">1000 genomes project</a></li>
</ul>
<p><br /></p>
<h2 id="preparation">Preparation</h2>
<h4 id="installation">Installation</h4>
<p>Install the released version by <code>pip</code> (Only support Python3.6+):</p>
<pre><code class="lang-bash">pip install GDBIGtools
# or upgrade
pip install --upgrade GDBIGtools
</code></pre>
<h4 id="help">Help</h4>
<p>type <code>GDBIGtools -h/--help</code> for detail.</p>
<pre><code>Usage: GDBIGtools [OPTIONS] COMMAND [ARGS]...

Options:
  -h, --help  Show this message and exit.

Commands:
  annotate   Annotate input VCF file with BIGCS allele Frequency.
  login      Login GDBIG.
  logout     Logout GDBIG.
  print-api  Display API information for GDBIG.
  query      Query variants from GDBIG database.
  version    GDBIGtools version: 1.1.6
</code></pre><h4 id="login">Login</h4>
<p>Login with <code>GDBIGtools</code> by using GDBIG API access key, which could be found from <a href="http://gdbig.bigcs.com.cn/api.html"><strong><API ACCESS\></strong></a> if you have apply for it.</p>
<p><a href="assets/figures/GDBIG_genomics_api.png"><img src="assets/figures/GDBIG_genomics_api.png" alt="GDBIG_genomics_api"></a></p>
<p>*Reset your GDBIG API access key if it had been leaked out.</p>
<p>And then you can login in command:</p>
<pre><code class="lang-bash">GDBIGtools login -k &lt;API Key&gt; -s &lt;Secret Key&gt;
</code></pre>
<p>If everything goes smoothly, means you can use GDBIG as one of your varaints database in command line mode.</p>
<h4 id="logout">Logout</h4>
<p>Logout <code>GDBIGtools</code> by simply run the command below:</p>
<pre><code class="lang-bash">GDBIGtools logout
</code></pre>
<p><br /></p>
<h2 id="quick-start">Quick start</h2>
<p><strong>*Note</strong>: Please enable your API access in <a href="http://gdbig.bigcs.com.cn/api.html"><strong>GDBIG API</strong></a> before using <strong>GDBIGtools</strong>.</p>
<h4 id="query-a-single-variant">Query a single variant</h4>
<p>Variants could be retrieved from GDBIG by using <code>query</code>.</p>
<p>Run <code>GDBIGtools query -h/--help</code> to see all available options. There\&#39;re
two different ways to retrive variants.</p>
<p>One is to use <code>-s</code> parameters for variants on command, the other way uses <code>-l</code> for input-file.</p>
<p>Here are examples for quering varaints on command.</p>
<pre><code class="lang-bash">GDBIGtools query -s rs117518546
GDBIGtools query -s 21:9662064
GDBIGtools query -s 22:10577666-10581518
GDBIGtools query -s ENST00000269305
GDBIGtools query -s MTHFR
</code></pre>
<p>It\&#39;ll take about 1 minute to complete about 2,000 variants\&#39; sites. 
Then you will get 32 new fields with the information of GDBIG in VCF INFO column:</p>
<ul>
<li><code>AF_GDBIG</code>: Alternate Allele Frequencies in GDBIG;</li>
<li><code>AF_GDBIG_SouthChina</code>: Alternate Allele Frequencies from SouthChina region in GDBIG;</li>
<li><code>AF_GDBIG_CentralChina</code>: Alternate Allele Frequencies from CentralChina region in GDBIG;</li>
<li><code>AF_GDBIG_EastChina</code>: Alternate Allele Frequencies from EastChina region in GDBIG.</li>
<li><code>AF_GDBIG_SouthwestChina</code>: Alternate Allele Frequencies from SouthwestChina region in GDBIG;</li>
<li><code>AF_GDBIG_NortheastChina</code>: Alternate Allele Frequencies from NortheastChina region in GDBIG;</li>
<li><code>AF_GDBIG_NorthwestChina</code>: Alternate Allele Frequencies from NorthwestChina region in GDBIG;</li>
<li><code>AF_GDBIG_NorthChina</code>: Alternate Allele Frequencies from NorthChina region in GDBIG;</li>
<li><code>AF_CMDB</code>: Alternate Allele Frequencies in CMDB database;</li>
<li><code>AF_ChinaMAP</code>: Alternate Allele Frequencies in ChinaMAP database;</li>
<li><code>AF_NyuWa</code>: Alternate Allele Frequencies in NyuWa(NCVD) database;</li>
<li><code>AF_WBBC</code>: Alternate Allele Frequencies in WBBC database;</li>
<li><code>AF_WBBC_North</code>: Alternate Allele Frequencies from North region in WBBC database;</li>
<li><code>AF_WBBC_Central</code>: Alternate Allele Frequencies from Central region in WBBC database;</li>
<li><code>AF_WBBC_South</code>: Alternate Allele Frequencies from South region in WBBC database;</li>
<li><code>AF_WBBC_Lingnan</code>: Alternate Allele Frequencies from Lingnan region in WBBC database;</li>
<li><code>AF_gnomAD</code>: Alternate Allele Frequencies in gnomAD database;</li>
<li><code>AF_gnomAD_afr</code>: Alternate Allele Frequencies from African-American/African population in gnomAD database;</li>
<li><code>AF_gnomAD_ami</code>: Alternate Allele Frequencies from Amish population in gnomAD database;</li>
<li><code>AF_gnomAD_asj</code>: Alternate Allele Frequencies from Ashkenazi Jewish population in gnomAD database;</li>
<li><code>AF_gnomAD_eas</code>: Alternate Allele Frequencies from East Asian population in gnomAD database;</li>
<li><code>AF_gnomAD_fin</code>: Alternate Allele Frequencies from Finnish population in gnomAD database;</li>
<li><code>AF_gnomAD_amr</code>: Alternate Allele Frequencies from Latino population in gnomAD database;</li>
<li><code>AF_gnomAD_nfe</code>: Alternate Allele Frequencies from Non-Finnish European population in gnomAD database;</li>
<li><code>AF_gnomAD_sas</code>: Alternate Allele Frequencies from South Asian population in gnomAD database;</li>
<li><code>AF_gnomAD_oth</code>: Alternate Allele Frequencies from Other population in gnomAD database;</li>
<li><code>AF_1KGP</code>: Alternate Allele Frequencies in 1000 Genomes database;</li>
<li><code>AF_1KGP_AFR</code>: Alternate Allele Frequencies from African population in 1000 Genomes database;</li>
<li><code>AF_1KGP_AMR</code>: Alternate Allele Frequencies from Admixed American population in 1000 Genomes database;</li>
<li><code>AF_1KGP_EAS</code>: Alternate Allele Frequencies from East Asian population in 1000 Genomes database;</li>
<li><code>AF_1KGP_EUR</code>: Alternate Allele Frequencies from European population in 1000 Genomes database;</li>
<li><code>AF_1KGP_SAS</code>: Alternate Allele Frequencies from South Asian population in 1000 Genomes database;</li>
</ul>
<p>And you will get query result looks like below:</p>
<pre><code class="lang-bash">##fileformat=VCFv4.2
##FILTER=&lt;ID=PASS,Description=&quot;All filters passed&quot;&gt;
##INFO=&lt;ID=AF_GDBIG,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_SouthChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from SouthChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_CentralChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from CentralChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_EastChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from EastChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_SouthwestChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from SouthwestChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_NortheastChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from NortheastChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_NorthwestChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from NorthwestChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_NorthChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from NorthChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_CMDB,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in CMDB database&quot;&gt;
##INFO=&lt;ID=AF_ChinaMAP,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in ChinaMAP database&quot;&gt;
##INFO=&lt;ID=AF_NyuWa,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in NyuWa(NCVD) database&quot;&gt;
##INFO=&lt;ID=AF_WBBC,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_WBBC_North,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from North region in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_WBBC_Central,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Central region in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_WBBC_South,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from South region in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_WBBC_Lingnan,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Lingnan region in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_afr,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from African-American/African population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_ami,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Amish population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_asj,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Ashkenazi Jewish population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_eas,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from East Asian population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_fin,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Finnish population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_amr,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Latino population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_nfe,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Non-Finnish European population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_sas,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from South Asian population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_oth,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Other population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_1KGP,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_AFR,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from African population in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_AMR,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Admixed American population in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_EAS,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from East Asian population in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_EUR,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from European population in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_SAS,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from South Asian population in 1000 Genomes database&quot;&gt;
##reference=file:/human_reference/GRCh38/GCA_000001405.15_GRCh38_no_alt_analysis_set.fa
#CHROM  POS ID  REF ALT QUAL  FILTER  INFO
chr22 10577666  rs1491296197  CAT C . PASS  AF_GDBIG=0.000123;AF_GDBIG_SouthChina=0.000169;AF_GDBIG_CentralChina=0;AF_GDBIG_EastChina=0;AF_GDBIG_SouthwestChina=0;AF_GDBIG_NortheastChina=0;AF_GDBIG_NorthwestChina=0;AF_GDBIG_NorthChina=0;AF_CMDB=NA;AF_ChinaMAP=0.000605;AF_NyuWa=NA;AF_WBBC=0.00033415;AF_WBBC_North=0.00037092;AF_WBBC_Central=0;AF_WBBC_South=0.00037092;AF_WBBC_Lingnan=0;AF_gnomAD=0.000224749;AF_gnomAD_afr=9.27558e-05;AF_gnomAD_ami=0;AF_gnomAD_asj=0;AF_gnomAD_eas=0.00037679;AF_gnomAD_fin=0;AF_gnomAD_amr=0.000986193;AF_gnomAD_nfe=0.000206587;AF_gnomAD_sas=0;AF_gnomAD_oth=0;AF_1KGP=NA;AF_1KGP_AFR=NA;AF_1KGP_AMR=NA;AF_1KGP_EAS=NA;AF_1KGP_EUR=NA;AF_1KGP_SAS=NA
chr22 10577851  . TA  T . PASS  AF_GDBIG=0.000123;AF_GDBIG_SouthChina=0.000169;AF_GDBIG_CentralChina=0;AF_GDBIG_EastChina=0;AF_GDBIG_SouthwestChina=0;AF_GDBIG_NortheastChina=0;AF_GDBIG_NorthwestChina=0;AF_GDBIG_NorthChina=0;AF_CMDB=NA;AF_ChinaMAP=NA;AF_NyuWa=NA;AF_WBBC=NA;AF_WBBC_North=NA;AF_WBBC_Central=NA;AF_WBBC_South=NA;AF_WBBC_Lingnan=NA;AF_gnomAD=NA;AF_gnomAD_afr=NA;AF_gnomAD_ami=NA;AF_gnomAD_asj=NA;AF_gnomAD_eas=NA;AF_gnomAD_fin=NA;AF_gnomAD_amr=NA;AF_gnomAD_nfe=NA;AF_gnomAD_sas=NA;AF_gnomAD_oth=NA;AF_1KGP=NA;AF_1KGP_AFR=NA;AF_1KGP_AMR=NA;AF_1KGP_EAS=NA;AF_1KGP_EUR=NA;AF_1KGP_SAS=NA
chr22 10580900  . ATTC  A . PASS  AF_GDBIG=0.000369;AF_GDBIG_SouthChina=0.000506;AF_GDBIG_CentralChina=0;AF_GDBIG_EastChina=0;AF_GDBIG_SouthwestChina=0;AF_GDBIG_NortheastChina=0;AF_GDBIG_NorthwestChina=0;AF_GDBIG_NorthChina=0;AF_CMDB=NA;AF_ChinaMAP=NA;AF_NyuWa=NA;AF_WBBC=NA;AF_WBBC_North=NA;AF_WBBC_Central=NA;AF_WBBC_South=NA;AF_WBBC_Lingnan=NA;AF_gnomAD=NA;AF_gnomAD_afr=NA;AF_gnomAD_ami=NA;AF_gnomAD_asj=NA;AF_gnomAD_eas=NA;AF_gnomAD_fin=NA;AF_gnomAD_amr=NA;AF_gnomAD_nfe=NA;AF_gnomAD_sas=NA;AF_gnomAD_oth=NA;AF_1KGP=NA;AF_1KGP_AFR=NA;AF_1KGP_AMR=NA;AF_1KGP_EAS=NA;AF_1KGP_EUR=NA;AF_1KGP_SAS=NA
chr22 10581005  rs1268262722  C T . PASS  AF_GDBIG=0.000123;AF_GDBIG_SouthChina=0;AF_GDBIG_CentralChina=0;AF_GDBIG_EastChina=0;AF_GDBIG_SouthwestChina=0.003571;AF_GDBIG_NortheastChina=0;AF_GDBIG_NorthwestChina=0;AF_GDBIG_NorthChina=0;AF_CMDB=NA;AF_ChinaMAP=9.9e-05;AF_NyuWa=NA;AF_WBBC=NA;AF_WBBC_North=NA;AF_WBBC_Central=NA;AF_WBBC_South=NA;AF_WBBC_Lingnan=NA;AF_gnomAD=2.39067e-05;AF_gnomAD_afr=0;AF_gnomAD_ami=0;AF_gnomAD_asj=0;AF_gnomAD_eas=0;AF_gnomAD_fin=0;AF_gnomAD_amr=0;AF_gnomAD_nfe=5.56091e-05;AF_gnomAD_sas=0;AF_gnomAD_oth=0;AF_1KGP=NA;AF_1KGP_AFR=NA;AF_1KGP_AMR=NA;AF_1KGP_EAS=NA;AF_1KGP_EUR=NA;AF_1KGP_SAS=NA
</code></pre>
<h4 id="quering-for-input-file">Quering for input-file</h4>
<p>A list of variants could be retrieved from GDBIG by using the parameters of <code>-l</code> when apply by <code>query</code>.</p>
<pre><code class="lang-bash">GDBIGtools query -l positions.list &gt; result.vcf
</code></pre>
<p>Format for <a href="tests/positions.list">positions.list</a>, could be a mixture of</p>
<ul>
<li><code>rs ID</code></li>
<li><code>ensembl transcript ID</code></li>
<li><code>gene symbol</code> and <code>ensembl gene ID</code></li>
<li><code>chrom   position</code> and <code>chrom    start   end</code>, even with or without <code>chr</code> in the chromosome ID column</li>
</ul>
<pre><code>#search key words
rs117518546
chr1:11795125
ENST00000269305
MTHFR  # fetch the best transcript annotaion results

#CHROM    POS    [POS_END]
chr22    17662883
22    17669209    17669357
</code></pre><p>Actrually you can use <code>-s</code> and <code>-l</code> simultaneously if you want.</p>
<pre><code class="lang-bash">GDBIGtools query -s 22:46616520 -l positions.list &gt; result.vcf
</code></pre>
<h4 id="annotate-your-vcf-files">Annotate your VCF files</h4>
<p>Annotate your VCF file with GDBIG by using <code>GDBIGtools annotate</code> command.</p>
<p>Download a list of example variants in VCF format from <a href="tests/GDBIG.test.vcf">GDBIG.test.vcf</a>. 
To annotate this list of variants with allele frequences from GDBIG, you can just run the following command in Linux or Mac OS.</p>
<pre><code class="lang-bash">GDBIGtools annotate -i GDBIG.test.vcf | bgzip &gt; output.GDBIG.test.vcf.gz
</code></pre>
<p><code>output.GDBIG.test.vcf.gz</code> is VCF format and looks like below:</p>
<pre><code class="lang-bash">##fileformat=VCFv4.2
##FILTER=&lt;ID=PASS,Description=&quot;All filters passed&quot;&gt;
##FORMAT=&lt;ID=DS,Number=1,Type=Float,Description=&quot;estimated ALT dose [P(RA) + P(AA)]&quot;&gt;
##FORMAT=&lt;ID=GP,Number=G,Type=Float,Description=&quot;Estimated Genotype Probability&quot;&gt;
##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
##bcftools_concatVersion=1.9+htslib-1.9
##reference=file:/GRCh38/GCA_000001405.15_GRCh38_no_alt_analysis_set.fa
##INFO=&lt;ID=AF_GDBIG,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_SouthChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from SouthChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_CentralChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from CentralChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_EastChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from EastChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_SouthwestChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from SouthwestChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_NortheastChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from NortheastChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_NorthwestChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from NorthwestChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_GDBIG_NorthChina,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from NorthChina region in GDBIG_GRCh38_v1.0&quot;&gt;
##INFO=&lt;ID=AF_CMDB,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in CMDB database&quot;&gt;
##INFO=&lt;ID=AF_ChinaMAP,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in ChinaMAP database&quot;&gt;
##INFO=&lt;ID=AF_NyuWa,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in NyuWa(NCVD) database&quot;&gt;
##INFO=&lt;ID=AF_WBBC,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_WBBC_North,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from North region in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_WBBC_Central,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Central region in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_WBBC_South,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from South region in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_WBBC_Lingnan,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Lingnan region in WBBC database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_afr,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from African-American/African population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_ami,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Amish population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_asj,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Ashkenazi Jewish population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_eas,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from East Asian population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_fin,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Finnish population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_amr,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Latino population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_nfe,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Non-Finnish European population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_sas,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from South Asian population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_gnomAD_oth,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Other population in gnomAD database&quot;&gt;
##INFO=&lt;ID=AF_1KGP,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_AFR,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from African population in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_AMR,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from Admixed American population in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_EAS,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from East Asian population in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_EUR,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from European population in 1000 Genomes database&quot;&gt;
##INFO=&lt;ID=AF_1KGP_SAS,Number=A,Type=Float,Description=&quot;Alternate Allele Frequencies from South Asian population in 1000 Genomes database&quot;&gt;
#CHROM  POS ID  REF ALT QUAL  FILTER  INFO
chr22 10515882  rs1490973086  G A . PASS  AR2=0.63;DR2=0.68;AF_GDBIG=0.105296;AF_GDBIG_SouthChina=0.106336;AF_GDBIG_CentralChina=0.116307;AF_GDBIG_EastChina=0.113487;AF_GDBIG_SouthwestChina=0.078571;AF_GDBIG_NortheastChina=0.098837;AF_GDBIG_NorthwestChina=0.061644;AF_GDBIG_NorthChina=0.057377;AF_CMDB=NA;AF_ChinaMAP=NA;AF_NyuWa=NA;AF_WBBC=NA;AF_WBBC_North=NA;AF_WBBC_Central=NA;AF_WBBC_South=NA;AF_WBBC_Lingnan=NA;AF_gnomAD=0.0918221;AF_gnomAD_afr=0.0347594;AF_gnomAD_ami=0.123162;AF_gnomAD_asj=0.196657;AF_gnomAD_eas=0.261649;AF_gnomAD_fin=0.171307;AF_gnomAD_amr=0.0983247;AF_gnomAD_nfe=0.10664;AF_gnomAD_sas=0.20235;AF_gnomAD_oth=0.11147;AF_1KGP=NA;AF_1KGP_AFR=NA;AF_1KGP_AMR=NA;AF_1KGP_EAS=NA;AF_1KGP_EUR=NA;AF_1KGP_SAS=NA
chr22 10516264  . TAC T . PASS  AR2=0.78;DR2=0.78;AF_GDBIG=0.000123;AF_GDBIG_SouthChina=0;AF_GDBIG_CentralChina=0;AF_GDBIG_EastChina=0.001645;AF_GDBIG_SouthwestChina=0;AF_GDBIG_NortheastChina=0;AF_GDBIG_NorthwestChina=0;AF_GDBIG_NorthChina=0;AF_CMDB=NA;AF_ChinaMAP=NA;AF_NyuWa=NA;AF_WBBC=NA;AF_WBBC_North=NA;AF_WBBC_Central=NA;AF_WBBC_South=NA;AF_WBBC_Lingnan=NA;AF_gnomAD=NA;AF_gnomAD_afr=NA;AF_gnomAD_ami=NA;AF_gnomAD_asj=NA;AF_gnomAD_eas=NA;AF_gnomAD_fin=NA;AF_gnomAD_amr=NA;AF_gnomAD_nfe=NA;AF_gnomAD_sas=NA;AF_gnomAD_oth=NA;AF_1KGP=NA;AF_1KGP_AFR=NA;AF_1KGP_AMR=NA;AF_1KGP_EAS=NA;AF_1KGP_EUR=NA;AF_1KGP_SAS=NA
chr22 10516615  rs1228174166  TTTG  T . PASS  AR2=0.221;DR2=0.222
chr22 10518420  rs1177693979  CA  C . PASS  AR2=0.547;DR2=0.607;AF_GDBIG=0.000246;AF_GDBIG_SouthChina=0.000337;AF_GDBIG_CentralChina=0;AF_GDBIG_EastChina=0;AF_GDBIG_SouthwestChina=0;AF_GDBIG_NortheastChina=0;AF_GDBIG_NorthwestChina=0;AF_GDBIG_NorthChina=0;AF_CMDB=NA;AF_ChinaMAP=0.00069;AF_NyuWa=NA;AF_WBBC=NA;AF_WBBC_North=NA;AF_WBBC_Central=NA;AF_WBBC_South=NA;AF_WBBC_Lingnan=NA;AF_gnomAD=NA;AF_gnomAD_afr=NA;AF_gnomAD_ami=NA;AF_gnomAD_asj=NA;AF_gnomAD_eas=NA;AF_gnomAD_fin=NA;AF_gnomAD_amr=NA;AF_gnomAD_nfe=NA;AF_gnomAD_sas=NA;AF_gnomAD_oth=NA;AF_1KGP=NA;AF_1KGP_AFR=NA;AF_1KGP_AMR=NA;AF_1KGP_EAS=NA;AF_1KGP_EUR=NA;AF_1KGP_SAS=NA
chr22 10519243  rs1186022611  A AG  . PASS  AR2=0.673;DR2=0.707;AF_GDBIG=0.000123;AF_GDBIG_SouthChina=0.000169;AF_GDBIG_CentralChina=0;AF_GDBIG_EastChina=0;AF_GDBIG_SouthwestChina=0;AF_GDBIG_NortheastChina=0;AF_GDBIG_NorthwestChina=0;AF_GDBIG_NorthChina=0;AF_CMDB=NA;AF_ChinaMAP=0.000446;AF_NyuWa=NA;AF_WBBC=NA;AF_WBBC_North=NA;AF_WBBC_Central=NA;AF_WBBC_South=NA;AF_WBBC_Lingnan=NA;AF_gnomAD=2.60797e-05;AF_gnomAD_afr=0;AF_gnomAD_ami=0;AF_gnomAD_asj=0;AF_gnomAD_eas=0.000484966;AF_gnomAD_fin=0;AF_gnomAD_amr=0.000154607;AF_gnomAD_nfe=0;AF_gnomAD_sas=0;AF_gnomAD_oth=0;AF_1KGP=NA;AF_1KGP_AFR=NA;AF_1KGP_AMR=NA;AF_1KGP_EAS=NA;AF_1KGP_EUR=NA;AF_1KGP_SAS=NA
</code></pre>
<p><br /></p>
<h2 id="citation">Citation</h2>
<p>-</p>

          	</article>
        </div>
		</div>
  </body>
</html>
<script type="text/javascript" src="toc/js/jquery-1.4.4.min.js"></script>
<script type="text/javascript" src="toc/js/jquery.ztree.all-3.5.min.js"></script>
<script type="text/javascript" src="toc/js/ztree_toc.js"></script>
<script type="text/javascript" src="toc_conf.js"></script>

<SCRIPT type="text/javascript" >
<!--
$(document).ready(function(){
    var css_conf = eval(markdown_panel_style);
    $('#readme').css(css_conf)
    
    var conf = eval(jquery_ztree_toc_opts);
		$('#tree').ztree_toc(conf);
});
//-->
</SCRIPT>